File: src/CalcViewModel/Common/CalculatorButtonPressedEventArgs.cpp
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

#include "pch.h"
#include "CalculatorButtonPressedEventArgs.h"

using namespace CalculatorApp;
using namespace CalculatorApp::ViewModel;
using namespace CalculatorApp::ViewModel::Common;
using namespace Platform;

NumbersAndOperatorsEnum CalculatorButtonPressedEventArgs::GetOperationFromCommandParameter(_In_ Object ^ commandParameter)
{
    auto eventArgs = dynamic_cast<CalculatorButtonPressedEventArgs ^>(commandParameter);
    if (eventArgs != nullptr)
    {
        return eventArgs->Operation;
    }
    else
    {
        return safe_cast<NumbersAndOperatorsEnum>(commandParameter);
    }
}

String ^ CalculatorButtonPressedEventArgs::GetAuditoryFeedbackFromCommandParameter(_In_ Object ^ commandParameter)
{
    auto eventArgs = dynamic_cast<CalculatorButtonPressedEventArgs ^>(commandParameter);
    if (eventArgs != nullptr)
    {
        return eventArgs->AuditoryFeedback;
    }
    else
    {
        return nullptr;
    }
}

File: src/Calculator.ViewModels/Common/CalculatorButtonPressedEventArgs.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

namespace CalculatorApp.ViewModel.Common
{
    public sealed class CalculatorButtonPressedEventArgs
    {
        public string AuditoryFeedback { get; }
        public NumbersAndOperatorsEnum Operation { get; }

        public CalculatorButtonPressedEventArgs(string auditoryFeedback, NumbersAndOperatorsEnum operation)
        {
            AuditoryFeedback = auditoryFeedback;
            Operation = operation;
        }

        public static NumbersAndOperatorsEnum GetOperationFromCommandParameter(object commandParameter)
        {
            if (commandParameter is CalculatorButtonPressedEventArgs eventArgs)
            {
                return eventArgs.Operation;
            }
            return (NumbersAndOperatorsEnum)commandParameter;
        }

        public static string GetAuditoryFeedbackFromCommandParameter(object commandParameter)
        {
            if (commandParameter is CalculatorButtonPressedEventArgs eventArgs)
            {
                return eventArgs.AuditoryFeedback;
            }
            return string.Empty;
        }
    }
}
